src/data/redux.test.jsx

```diff
@@ -664,14 +679,61 @@ describe('Data layer integration tests', () => {
     });
 
     it('Should call the exams service for update attempt', async () => {
-      const updateExamAttemptURL = `${getConfig().EXAMS_BASE_URL}/attempt/${attempt.attempt_id}`;
+      const updateExamAttemptURL = `${getConfig().EXAMS_BASE_URL}/api/v1/attempt/${attempt.attempt_id}`;
+      const examURL = `${getConfig().EXAMS_BASE_URL}/api/v1/student/exam/attempt/course_id/${courseId}/content_id/${contentId}`;
+      const activeAttemptURL = `${getConfig().EXAMS_BASE_URL}/api/v1/exams/attempt/latest`;
 
-      axiosMock.onGet(fetchExamAttemptsDataUrl).replyOnce(200, { exam: {}, active_attempt: attempt });
+      axiosMock.onGet(examURL).reply(200, { exam });
+      axiosMock.onGet(activeAttemptURL).reply(200, { attempt });
       axiosMock.onPut(updateExamAttemptURL).reply(200, { exam_attempt_id: attempt.attempt_id });
 
       await executeThunk(thunks.getExamAttemptsData(courseId, contentId), store.dispatch);
       await executeThunk(thunks.stopExam(), store.dispatch, store.getState);
       expect(axiosMock.history.put[0].url).toEqual(updateExamAttemptURL);
     });
+
+    it('Should call the exams service to fetch attempt data', async () => {
+      const examURL = `${getConfig().EXAMS_BASE_URL}/api/v1/student/exam/attempt/course_id/${courseId}/content_id/${contentId}`;
+      const activeAttemptURL = `${getConfig().EXAMS_BASE_URL}/api/v1/exams/attempt/latest`;
+
+      axiosMock.onGet(examURL).reply(200, { exam });
+      axiosMock.onGet(activeAttemptURL).reply(200, { attempt });
+
+      await executeThunk(thunks.getExamAttemptsData(courseId, contentId), store.dispatch);
+
+      expect(axiosMock.history.get[0].url).toEqual(examURL);
+      expect(axiosMock.history.get[1].url).toEqual(activeAttemptURL);
+
+      const state = store.getState();
+      expect(state).toMatchSnapshot();
+    });
+
+    it('Should call the exams service to poll/refresh latest attempt data', async () => {
+      // Mock the endpoints that will be called on edx-exams
+      const examURL = `${getConfig().EXAMS_BASE_URL}/api/v1/student/exam/attempt/course_id/${courseId}/content_id/${contentId}`;
+      const activeAttemptURL = `${getConfig().EXAMS_BASE_URL}/api/v1/exams/attempt/latest`;
+
+      axiosMock.onGet(examURL).reply(200, { exam });
+      axiosMock.onGet(activeAttemptURL).reply(200, { attempt });
+
+      // make sure that the state is initialized by calling "getExamAttemptsData" first
+      await executeThunk(thunks.getExamAttemptsData(courseId, contentId), store.dispatch);
```

Comment:
getExamAttemptsData is also calling the `activeAttemptUrl`, so I would suggest mocking out two calls to that endpoint: one for the initial state, one to check if things have been updated. That way you can verify that the data has been updated in the snapshot correctly

Reply:
Hi there,

I figured out how to implement what you said, and ran into a couple of hiccups.

When only executing the `pollAttempt` thunk, the test ends up saving a snapshot of an exam/attempt object with a bunch of empty and null fields, with the apiErrorMsg field set to `Cannot set properties of undefined (setting 'active_attempt')`.

From what I can tell, this may be due to this line of code at the very start of the `pollAttempt` thunk:
```const currentAttempt = getState().examState.activeAttempt;```
If my understanding is correct, the `pollAttempt` thunk requires the active attempt in the UI's state to be initialized before being called.

As such, in my latest commit, the test calls the `getLatestAttemptData` to initialize the data. The attempt that is returned does then have some data filled out, though it still does contain the same apiErrorMsg, alongside some empty fields.

Reply:
Yeah that makes sense, so the state needs to be initialized prior to polling the attempt. 

If I'm understanding correctly, your current test has `getLatestAttemptData` and `pollAttempt` returning the same data (via the URL mocks). Because of that, it's not clear in the test if the data is actually being updated as expected via the poll attempt URL. 

It's possible to return two separate data responses for the same URL (if you have two different mocks, there are a few examples of that in this testing file). That way you can verify via the snapshot that the poll attempt URL is updating the state. 

Let me know if I'm missing something in my understanding though!

Reply:
If you're not calling `getLatestAttemptsData`, can you restore the original test then? Just to make sure that the thunk is being tested.